lib/types.nix

```diff
@@ -498,7 +498,7 @@ rec {
       name = "pathInStore";
       description = "path in the Nix store";
       descriptionClass = "noun";
-      check = x: isStringLike x && builtins.match "${builtins.storeDir}/[^.].*" (toString x) != null;
+      check = x: (isStringLike x && builtins.match "${builtins.storeDir}/[^.].*" (toString x) != null) || lib.isDerivation x;
```

Comment:
Regex for here is a bit more complicated so help needed

Here's a tester https://regex101.com/r/qGlRY9/1